dbms/src/Functions/FunctionsString.cpp

```diff
@@ -4490,30 +4494,31 @@ class FunctionASCII : public IFunction
     void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
     {
         const IColumn * c0_col = block.getByPosition(arguments[0]).column.get();
-        const auto * c0_const = checkAndGetColumn<ColumnConst>(c0_col);
         const auto * c0_string = checkAndGetColumn<ColumnString>(c0_col);
-
-        Field res_field;
-        int val_num = c0_col->size();
-        auto col_res = ColumnInt64::create();
-        col_res->reserve(val_num);
-        if (c0_const == nullptr && c0_string == nullptr)
+        if unlikely (c0_string == nullptr)
             throw Exception(
                 fmt::format("Illegal argument of function {}", getName()),
                 ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
 
-        for (int i = 0; i < val_num; i++)
-        {
-            c0_col->get(i, res_field);
-            String handled_str = res_field.get<String>();
-            Int64 res = handled_str.empty() ? 0 : static_cast<Int64>(handled_str[0]);
-            col_res->insert(res);
-        }
+        size_t val_num = c0_col->size();
+        auto col_res = ColumnInt64::create();
+        col_res->reserve(val_num);
+
+        const auto & chars = c0_string->getChars();
+        const auto & offsets = c0_string->getOffsets();
+
+        if (val_num > 0)
+            col_res->insert(getResult(chars, 0));
+        
+
+        for (size_t i = 1; i < val_num; i++)
+            col_res->insert(getResult(chars, offsets[i - 1]));
 
         block.getByPosition(result).column = std::move(col_res);
     }
 
 private:
+static Int64 getResult(const ColumnString::Chars_t & chars, size_t offset) { return chars[offset]; }
```

Comment:
The logic is straightforward, there's no need to add a function.

Reply:
> The logic is straightforward, there's no need to add a function.

done